net/dns/resolver/tsdns.go

```diff
@@ -596,11 +608,23 @@ func (r *Resolver) resolveLocal(domain dnsname.FQDN, typ dns.Type) (netip.Addr,
 
 	r.mu.Lock()
 	hosts := r.hostToIP
+	wildcards := r.wildcards
 	localDomains := r.localDomains
 	r.mu.Unlock()
 
-	addrs, found := hosts[domain]
-	if !found {
+	addrs, ok := hosts[domain]
+	if !ok {
+		labels := strings.Split(string(domain), ".")
```

Comment:
strings.Split allocates. As does strings.Join.

We should be able to do this by looking for dots and slicing the input string?

Add an allocation test too with testing.AllocsPerRun?